### DISCUSSION TITLE
How to build xyz for Android?

### DISCUSSION THREAD

**Discussion by Stingaaa** [cogentcore/core#1296] [Q&A] [answered]
When I try to build the xyz example for android I get the following error:
`# cogentcore.org/core/xyz/io/obj
..\..\..\..\go\pkg\mod\cogentcore.org\core@v0.3.5\xyz\io\obj\obj.go:296:26: invINDEX (untyped int constant 4294967295) overflows int
..\..\..\..\go\pkg\mod\cogentcore.org\core@v0.3.5\xyz\io\obj\obj.go:305:22: invINDEX (untyped int constant 4294967295) overflows int
..\..\..\..\go\pkg\mod\cogentcore.org\core@v0.3.5\xyz\io\obj\obj.go:584:20: cannot use invINDEX (untyped int constant 4294967295) as int value in assignment (overflows)
..\..\..\..\go\pkg\mod\cogentcore.org\core@v0.3.5\xyz\io\obj\obj.go:606:24: cannot use invINDEX (untyped int constant 4294967295) as int value in assignment (overflows)
core build failed: failed to run "go build -buildmode=c-shared -ldflags -X cogentcore.org/core/system.CoreVersion=v0.3.5 -o C:\Users\maxip\AppData\Local\Temp\gomobile-work-3747029769\lib\x86\libDesktop.so BikeTUNE.at/cmd/desktop: exit status 1"`

When I looked deeper into the error itself, I found out that the invINDEX is a uint32, which of course will overflow a 32-bit integer, therefore I am wondering if I need to install something, which lets me build for Android (/mobile in general)?
(I read in the xyz [README](https://github.com/cogentcore/core/blob/main/xyz/README.md) that it is compatible with mobile platforms which I assumed also includes Android)

**Comment by kkoreilly** [comment, marked as answer]
Thank you for reporting this. Xyz is compatible with Android and we will make it compatible with 32-bit soon. In the meantime, you need to build it specifically for a 64-bit Android platform due to that 32-bit overflow. For example, if you have an amd64 Android device, you could do this:

```sh
core build android/amd64
```

The fact that it does not work on 32-bit is not a fundamental limitation, just a minor error, so we will get it working on 32-bit soon. Apologies for that issue.

**Reply by Stingaaa** [reply]
As most of the times, it's the most obvious solution you don't see.
Thank you very much for the reply.

**Reply by kkoreilly** [reply]
I fixed the underlying issue with 32-bit platforms in #1297, so once that is merged there should be no problems with building xyz for any platforms.
